### ISSUE TITLE
query_yn in crafting.cpp - unexpected color behavior

### ISSUE THREAD

**Issue by Pupsi-Mupsi** [CleverRaven/Cataclysm-DDA#27474]
**Describe the bug**  
If a query contains an object with "healthbars" not only the name but all following characters are colored in white.

**Expected behavior**  
"may yield:" should not be in white.

**Screenshots**  
![grafik](https://user-images.githubusercontent.com/44737997/50743924-a4d0a700-121d-11e9-9fdf-3387eb0b35fe.png)

![grafik](https://user-images.githubusercontent.com/44737997/50743984-65568a80-121e-11e9-9403-f4b6026ade71.png)

**Comment by Pupsi-Mupsi** [comment]
BTW I was working on this - so I don't mind the name beeing in color...

```
    const auto &r = recipe_dictionary::get_uncraft( obj.typeId() );
    // last chance to back out
    if( interactive && get_option<bool>( "QUERY_DISASSEMBLE" ) ) {
        const auto components( r.disassembly_requirements().get_components() );
        std::ostringstream list;
        for( const auto &component : components ) {
            list << "<color_white>-</color> " << "<color_light_gray>" << component.front().to_string() << "</color>" << std::endl;
        }
```
![grafik](https://user-images.githubusercontent.com/44737997/50744038-10ffda80-121f-11e9-8c25-39a9c2e65e77.png)

**Comment by kevingranade** [comment]
Confirmed locally, *should* be relatively simple, just missing a closing tag or something.

**Comment by AMurkin** [comment]
It seems color have been cleared here:
https://github.com/CleverRaven/Cataclysm-DDA/blob/81e9990c2250e230c36c6d877c09ad6d701a4afc/src/item.cpp#L2817

**Comment by jbytheway** [comment]
I'm pretty sure it's because the colouring code doesn't handle nested colour tags.

**Comment by kevingranade** [comment]
Fixed by both #27838 and #27856
